### DISCUSSION TITLE
How to deactivate 2FA

### DISCUSSION THREAD

**Discussion by Flooxou** [gravitational/teleport#49838] [Q&A] [answered]
Hello,
Now with the new version of Teleport, it is possible to disable the 2FA. But I have the impression that it is not done as before version 16. Is it possible to document how the deactivate? and is it possible to do it with teleport as a package or do you need the docker version?

**Comment by webvictim** [comment]
You need to set the `TELEPORT_ALLOW_NO_SECOND_FACTOR` environment variable to `yes`, then restart Teleport.

Assuming you're running Teleport with systemd, this is most easily done like this:

```bash
echo "TELEPORT_ALLOW_NO_SECOND_FACTOR=yes" | sudo tee -a /etc/default/teleport
sudo systemctl restart teleport
```

For Kubernetes-based deployments using the `teleport-cluster` Helm chart, you'll need to add it to your chart values and then `helm upgrade`:

```yaml
extraEnv:
- name: TELEPORT_ALLOW_NO_SECOND_FACTOR
  value: yes
```

**Reply by Flooxou** [reply]
I tried this method but when I restart teleport, I always need to do the 2FA. Even if I try to create a new user. I am in version 17.0.3 however.

**Reply by zmb3** [reply, marked as answer]
It should work in version 17.0.3.

You'll need to set the env var **and** set `second_factor: off` in your configuration.

- Teleport 15 and earlier allow you to disable MFA without setting any env var.
- Teleport 16 requires you to set `TELEPORT_ALLOW_NO_SECOND_FACTOR=yes` in order to disable MFA.
- Teleport 17.0.1-17.0.2 provide no way to disable MFA.
- Teleport 17.0.3 restores the Teleport 16 behavior.

**Reply by Flooxou** [reply]
Ahhhh thank you, I hadn't put second_factor: off.

**Reply by manish-chalo** [reply]
Any way to disable mfa in version 18.3.2 or higher?

**Reply by Flooxou** [reply]
I'm still on version 1.17 and it continues to work without MFA. However, you need to be careful with the TELEPORT_ALLOW_NO_SECOND_FACTOR variable; you probably need to set it to “True.” Teleport is a little temperamental following an update.

**Reply by Flooxou** [reply]
For the second_factors: off to work, you must set the TELEPORT_ALLOW_NO_SECOND_FACTOR environment variable to True, like this: TELEPORT_ALLOW_NO_SECOND_FACTOR=True in your deployment

**Reply by diktamxx** [reply]
> For the second_factors: off to work, you must set the TELEPORT_ALLOW_NO_SECOND_FACTOR environment variable to True, like this: TELEPORT_ALLOW_NO_SECOND_FACTOR=True in your deployment

It seems that the setting in the version public.ecr.aws/gravitational/teleport-distroless:18.6.6 does not work.

I truly believe that a software program should be accountable not only for the production environment but also for the testing environment. If it is not easy to test, users will have difficulty in assessing whether they should adopt Teleport. I have seen many cases where users gave up using Teleport because of MFA...

**Reply by Flooxou** [reply]
I see what you mean, but in normal operation you need to have MFA, so it's "normal" that it's not easy to remove. Show me your Docker Compose file so I can see if I can reproduce your error.

**Comment by 13567436138** [comment]
^Croot@ubuntu:~# kubectl logs -n teleport teleport-auth-767ff767d8-5npbx 
ERROR: cannot disable multi-factor authentication

**Reply by 13567436138** [reply]
18

**Reply by Flooxou** [reply]
I need more details to understand why it's not working. Please provide your Kube manifest with the sensitive keys anonymized.
